src/db/commit.rs

```diff
@@ -121,6 +123,7 @@ impl Commit {
         meta.local_create_date().ok_or(MissingLocalCreateDate)?;
         // basis_hash is optional -- the first commit lacks a basis
         meta.checksum().ok_or(MissingChecksum)?;
+        Checksum::from_str(meta.checksum().unwrap()).map_err(|_| InvalidChecksum)?;
```

Comment:
Nit: eliminate the scary looking unwrap by storing the result of the previous statement and using that in place of `meta.checksum().unwrap()`.